### PR TITLE
external client FAQ update

### PR DESCRIPTION
**Pull request by tbarnes-us**
Update the external client FAQ to cover additional use cases:
- clients running in the same namespace as their target
- clients running in the same k8s cluster as their target, but in a different namespace

### CODE REVIEW

File: documentation/staging/content/faq/external-clients.md [hunk 1]
```diff
@@ -162,6 +244,37 @@ And, here is a snippet of offline WLST code that corresponds to the above `confi
   set('ClientCertificateEnforced', false)
 ```
 
+And, here is a snippet of WDT model YAML configuration that corresponds to the above `snippets`:
```
Comment (rosemarymarano) [inline documentation/staging/content/faq/external-clients.md]: And, here -> Here (you don't need the "And" in this or the above sentence)
YAML  -> YAML file

File: documentation/staging/content/faq/external-clients.md [hunk 2]
```diff
@@ -261,6 +374,7 @@ To enable an 'unknown host' source WebLogic Server to initiate EJB, JMS, or JTA
 
 #### Optional reading
 
-- For sample JMS client code and JMS configuration, see [Run Standalone WebLogic JMS Clients on Kubernetes](https://blogs.oracle.com/weblogicserver/run-standalone-weblogic-jms-clients-on-kubernetes).
+- For a discussion of the WebLogic URL syntax for JMS, EJB, and JNDI clients see [Understanding WebLogic URLs](https://docs.oracle.com/en/middleware/fusion-middleware/weblogic-server/12.2.1.4/jmsad/best_practice.html#GUID-CFA6D7B9-6059-4098-801C-30F0EE31C1AA).
```
Comment (rosemarymarano) [inline documentation/staging/content/faq/external-clients.md:377]: discussion  -> description (the referenced texts are descriptions not discussions; true for the following bulleted item as well)
For a detailed discussion -> For a detailed description

**Review by rosemarymarano — COMMENTED** [review]
A few minor edits.

File: documentation/staging/content/faq/external-clients.md [hunk 3]
```diff
@@ -27,15 +27,98 @@ description: "This FAQ describes approaches for giving external WebLogic clients
...
+    resource's Kubernetes ClusterIP service. To see the service DNS names
+    for a deployed domain, run `kubectl -n MYNS get services`.
+    Example URLs:
+    - For a target resource that's hosted anywhere
+      in WebLogic cluster `mycluster`
+      with domain UID `myuid` where the cluster is listening
+      on port `7001`: `t3://myuid-cluster-mycluster:7001`.
+    - For a target resource that's hosted in a non-clustered
+      WebLogic Server `myserver` that is part
+      of domain with a domain UID `myuid` here the server
```
[the remark this answers is not in the document]
Comment (tbarnes-us) [reply documentation/staging/content/faq/external-clients.md]: fixed

File: documentation/staging/content/faq/external-clients.md [hunk 4]
```diff
@@ -27,15 +27,98 @@ description: "This FAQ describes approaches for giving external WebLogic clients
...
+If a WebLogic EJB or JMS resource is located in the same
+Kubernetes cluster as an application that calls the resource, but
+the application and resource are in different Kubernetes namespaces, then:
+
+- The URL used by the application can begin with `t3` or `t3s`, and the
+  DNS address in the URL must be fully specified to differentiate the namespace.
+  - For example,
+    if the target resource is in WebLogic cluster `mycluster` with
+    domain UID `myuid` in namespace `myns`, and the cluster is
+    listening on port 8001, then the cluster service name
```
[the remark this answers is not in the document]
Comment (tbarnes-us) [reply documentation/staging/content/faq/external-clients.md]: fixed

File: documentation/staging/content/faq/external-clients.md [hunk 5]
```diff
@@ -27,15 +27,98 @@ description: "This FAQ describes approaches for giving external WebLogic clients
...
+      if the target resource is in WebLogic cluster `mycluster` that
+      is part of domain with a domain UID `myuid` running in domain
+      `myns`, then
+      the cluster service name will be `myuid-cluster-mycluster`
+      and the fully decorated name will be
+      `myuid-cluster-mycluster.myns.service.cluster.local`.
+  - Do not specify a public port. It should be the same as the
+    the channel's listen port, which is the default. This is because
+    there is no port mapping between namespaces.
+  - Do not enable 'outbound enabled'.
```
[the remark this answers is not in the document]
Comment (tbarnes-us) [reply documentation/staging/content/faq/external-clients.md]: fixed

File: documentation/staging/content/faq/external-clients.md [hunk 6]
```diff
@@ -27,15 +27,98 @@ description: "This FAQ describes approaches for giving external WebLogic clients
...
+  - There is no need to enable tunneling, and you can continue
+    to use URLs that begin with `t3` or `t3s`.
+  Application URLs must specify the channel's port.
+
+If a WebLogic EJB or JMS resource is hosted inside a Kubernetes
+cluster, but an application that calls the resource
+is is hosted outside of the Kubernetes cluster, then:
```
[the remark this answers is not in the document]
Comment (tbarnes-us) [reply documentation/staging/content/faq/external-clients.md]: fixed

File: documentation/staging/content/faq/external-clients.md (hunk 2 above)
Comment (tbarnes-us) [reply documentation/staging/content/faq/external-clients.md]: fixed

**Review by rosemarymarano — APPROVED** [review]

File: documentation/staging/content/faq/external-clients.md [hunk 7]
```diff
@@ -3,7 +3,7 @@ title: "External WebLogic clients"
 date: 2019-11-21T21:23:03Z
 draft: false
 weight: 11
-description: "This FAQ describes approaches for giving external WebLogic clients or servers access to a Kubernetes hosted WebLogic cluster JMS or EJBs, and for giving Kubernetes hosted clients or servers access to remotely hosted WebLogic JMS or EJBs."
+description: "This FAQ describes approaches for giving external WebLogic clients or servers access to a Kubernetes hosted WebLogic cluster JMS or EJBs, for giving Kubernetes hosted clients or servers access to remotely hosted WebLogic JMS or EJBs. This includes between different Kubernetes namespaces within the same Kubernentes cluster, between different Kubernetes clusters, and from a non-Kubernetes location into Kubernetes."
```
[the remark this answers is not in the document]
Comment (tbarnes-us) [reply documentation/staging/content/faq/external-clients.md]: fixed (rewrote both sentences)

File: documentation/staging/content/faq/external-clients.md (hunk 1 above)
Comment (tbarnes-us) [reply documentation/staging/content/faq/external-clients.md]: fixed

File: documentation/staging/content/faq/external-clients.md [hunk 8]
```diff
@@ -27,15 +27,98 @@ description: "This FAQ describes approaches for giving external WebLogic clients
...
+    domain UID `myuid` in namespace `myns`, and the cluster is
+    listening on port `8001`, then the cluster service name 
+    that is generated by the operator will
+    be `myuid-cluster-mycluster` and the full URL will be
+    `t3://myuid-cluster-mycluster.myns.service.cluster.local:8001`.
+
+- If the applications run within a WebLogic
+  Server JVM, then the WebLogic Servers that host the target
+  EJB or JMS resources need to be configured to
+  [enable unknown host access](#enabling-unknown-host-access).
```
Comment (alai8) [inline documentation/staging/content/faq/external-clients.md:76]: We may not need to enable unknown host access if the resource and application are in the same k8s cluster, and fully decorated DNS address is used in the public address.

Comment (tbarnes-us) [reply documentation/staging/content/faq/external-clients.md:76]: A customer recently told us on Slack that they needed to do this. I'll add the word "may" before need.

**Review by alai8 — APPROVED** [review]

File: documentation/staging/content/faq/external-clients.md (hunk 8 above)
Comment (tbarnes-us) [reply documentation/staging/content/faq/external-clients.md]: I've updated and moved the bullet.

**Review by alai8 — APPROVED** [review]

**Pull request merged**